### PR TITLE
[#228] Allow developers and teams to download revenue reports

### LINKED ISSUE #228: Provide usage details for both prepaid and postpaid developers
Neither prepaid or postpaid developers can see their current usage. 
Prepaid can see summary usage via "Prepaid balance" tab.
In Edge Monetization Reports, a Revenue type report will show both prepaid and postpaid usages, with or without developer filtering.

These reports can be created using Edge API with or without DevCriteria using for both prepaid and postpaid developers:
https://apidocs.apigee.com/monetize/apis/post/organizations/%7Borg_name%7D/%7Breport_type%7D
Or for a specific developer, via URL path segment:
https://apidocs.apigee.com/monetize/apis/post/organizations/%7Borg_name%7D/developers/%7Bdeveloper_id%7D/revenue-reports

The output is a CSV file.

This would be a great feature enhancement that many customers have asked for.

For example:
```
curl -n -X POST \
  https://api.enterprise.apigee.com/v1/mint/organizations/amer-demo13/revenue-reports \
  -H 'Content-Type: application/json' \
  -d '{
	"fromDate": "2020-03-01 00:00:00",
	"toDate": "2020-04-01 00:00:00",
	"showTxDetail": true,
	"showSummary": true,
	"devCriteria": [{
		"id": "18a91b9c-05bc-4801-a919-6e8945918b13",
		"orgId": "amer-demo13"
	}],
	"transactionTypes": [
		"PURCHASE",
        	"CHARGE",
        	"REFUND",
        	"CREDIT",
        	"SETUPFEES",
        	"TERMINATIONFEES",
        	"RECURRINGFEES"
	],
	"currencyOption": "LOCAL",
	"groupBy": [
		"PACKAGE",
		"PRODUCT",
		"DEVELOPER",
        	"APPLICATION",
        	"RATEPLAN"
	]
}'
```

Another example for same thing but different API
```
curl -n -X POST \
  https://api.enterprise.apigee.com/v1/mint/organizations/amer-demo13/developers/18a91b9c-05bc-4801-a919-6e8945918b13/revenue-reports \
  -H 'Content-Type: application/json' \
  -d '{
	"fromDate": "2020-03-01 00:00:00",
	"toDate": "2020-04-01 00:00:00",
	"showTxDetail": true,
	"showSummary": true,
	"transactionTypes": [
		"PURCHASE",
        	"CHARGE",
        	"REFUND",
        	"CREDIT",
        	"SETUPFEES",
        	"TERMINATIONFEES",
        	"RECURRINGFEES"
	],
	"currencyOption": "LOCAL",
	"groupBy": [
		"PACKAGE",
		"PRODUCT",
		"DEVELOPER",
        	"APPLICATION",
        	"RATEPLAN"
	]
}'
```

### PR DESCRIPTION
**Pull request by shadcn**
Fixes #228 

This PR adds functionality to allow developers and teams to generate (csv) revenue reports.

Download_Reports___Apigee

### CODE REVIEW

File: src/MonetizationInterface.php [hunk 1]
```diff
@@ -152,6 +153,23 @@ public function getSupportedCurrencies(): ?array;
...
+   *   The from month for the report.
+   * @param \DateTimeImmutable $to_date
+   *   The to month for the report.
+   * @param string $currency
+   *   The currency id. Example: usd.
+   *
+   * @return null|string
+   *   A CSV string of revenue report.
+   */
+  public function getRevenueReport(string $developer_id, \DateTimeImmutable $from_date, \DateTimeImmutable $to_date, string $currency): ?string;
```
Comment (arlina-espinoza) [inline src/MonetizationInterface.php]: We can't add new methods without breaking BC, although I totally feel this should go in the Monetization service class. This is something that we should discuss further with @cnovak , as the monetization service isn't really defining a contract, but it's more of a collection of utility functions, and I feel it'd be too much to release a new major version everytime we want to add to it.

**Review by arlina-espinoza — CHANGES_REQUESTED** [review]
Thanks @arshad. This is looking good, but when I test for a given user that I know has info for given dates, it says that "There are no revenue reports for the selected dates.".
The request and response logged using apigee_edge_debug:

```
POST /v1/mint/organizations/***organization***/revenue-reports HTTP/1.1
{"currCriteria":[{"id":"usd","orgId":"***organization***"}],"devCriteria":[{"id":"***email***","orgId":"***organization***"}],"endDate":"2020-04-01 07:00:00","fromDate":"2020-03-01 08:00:00","showRevSharePct":false,"showSummary":false,"showTxDetail":true,"showTxType":false}

HTTP/1.1 400 Bad Request
{
  "code" : "mint.toDateRequired",
  "message" : "To Date is required.",
  "contexts" : [ ],
  "cause" : {
    "message" : "To Date is required.",
    "contexts" : [ ]
  }
}
```

Changing `fromDate` to `toDate` in the PHP client library `Apigee\Edge\Api\Monetization\Structure\Reports\Criteria\RevenueReportCriteria` makes the report work correctly, so I'm filing an issue for this, and we'll need to require the latest version once we release that fix.

The other thing is what I mentioned about not being able to add methods to the interface without breaking changes. :(

File: src/MonetizationInterface.php (hunk 1 above)
Comment (shadcn) [reply src/MonetizationInterface.php]: Yeah I've been wondering about this too. The `MonetizationInterface` is just a bunch of methods right now, might need to break this into actual interfaces.

It looks like we need to start looking into 2.x branch soon.

**Comment by shadcn** [comment]
> Changing `fromDate` to `toDate` in the PHP client library `Apigee\Edge\Api\Monetization\Structure\Reports\Criteria\RevenueReportCriteria` makes the report work correctly, so I'm filing an issue for this, and we'll need to require the latest version once we release that fix.

Oh I created a PR for this https://github.com/apigee/apigee-client-php/pull/101 (I should've mentioned this in this PR. Sorry).

File: src/MonetizationInterface.php (hunk 1 above)
Comment (shadcn) [reply src/MonetizationInterface.php]: @arlina-espinoza I removed the new method from the interface and move it to the form for now. We can refactor this for `2.x`. Thank you.

**Review by arlina-espinoza — APPROVED** [review]
Thanks for the changes @arshad. LGTM :)

**Pull request merged**